test/join.jl

```diff
@@ -584,8 +584,13 @@ module TestJoin
                                     Name = ["John Doe", "Jane Doe", "Jane Doe", "Joe Blogs", missing],
                                     Job = ["Lawyer", "Doctor", "Florist", missing, "Farmer"],
                                     _merge = ["both", "both", "both", "left_only", "right_only"])
+
+        pre_join_name = copy(name)
```

Comment:
As noted in the other PR, why is this needed?

Reply:
Because `name` is being modified by `join`. that's the purpose of this patch. So to test and make sure that isn't happening in the future, we need a copy of `name` from before the join to compare to the post-join version of join.

Reply:
OK. These cases are difficult to decide: virtually all functions could modify their inputs, and we don't typically check this for all of them. Anyway if you prefer adding this please add a comment.

Reply:
My philosophy if that if a bug is found, when you patch it you add a test to ensure that the bug would be caught if it weren't fixed. Otherwise how do you know if your patch works? I'll add a comment.